keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -1284,6 +1421,85 @@ The field to which the CEL rule in its `x-kubernetes-validations` list must
 contain all fields referred by the CEL rule; so the CEL error logic can have to 
 be modified to also include the rule's original location.
 
+### Different Ratcheting Rule Per Value Validation
+
+An alternative to the weaker form ratcheting described in the proposal would be
+to apply the DeepEqual rule to a smaller projection of the object being validated.
+
+i.e. MaxLength would use projection `len(obj)` since it only looks at that value,
+and if `len(oldObj) == len(newObj)`, then we would allow ratcheting.
+
+The following projections were considered:
+
+| Field               | Type     | Projection                   
+|---------------------|----------|------------------------------
+| Format              | string   | Identity                     
+| Maximum             | *float64  | Identity                    
+| ExclusiveMaximum    | bool     | Identity                    
+| Minimum             | *float64  | Identity                    
+| ExclusiveMinimum    | bool     | Identity                    
+| MaxLength           | *int64   | len(obj) -> int             
+| MinLength           | *int64   | len(obj) -> int             
+| Pattern             | string   | Identity                  
+| MultipleOf          | *float64  | Identity                 
+| Enum                | []JSON   | Identity                  
+| Nullable            | bool     | Identity                  
+| MaxItems            | *int64   | len(obj) -> int             
+| MinItems            | *int64   | len(obj) -> int             
+| UniqueItems         | bool     | Identity       
+| MaxProperties       | *int64   | len(obj) -> int             
+| MinProperties       | *int64   | len(obj) -> int             
+| Required            | []string | Keys(obj) -> []string (also can choose to ratchet on per-key presence)       
```

Comment:
We need to carefully identify which of these properties have some not-orthogonal dimensions. Changing one of the property can affect another property and so they can't be treated individually unless they are fully orthogonal. 
Pattern, MinLength, MaxLength, Format and probably some others can all affect the less of a string, so we can't just say "hello -> hell" is fine because `MaxLength: 3`, and `hell` is closer to 3 than hello, because that also makes us further from an old value of `MinLength`.

Reply:
Agreed, will reframe the correctness discussion of this KEP around the requirement that anything ratcheted must pass the old schema